crates/core/src/kernel/snapshot/mod.rs

```diff
@@ -315,8 +315,8 @@ impl Snapshot {
         let stats_fields = if let Some(stats_cols) = self.table_config().stats_columns() {
             stats_cols
                 .iter()
-                .map(|col| match schema.field_with_name(col) {
-                    Ok(field) => match field.data_type() {
+                .map(|col| match schema.field(col) {
```

Comment:
good question, let me investigate.

Reply:
@alexwilcoxson-rel - stated some asking around if it would be legal to have a nested field defined in that property.

There are a couple of questions that also come to mind, and maybe you have an opinion? So using `.` as a separator is - AFAIK - just a convention (certainly a common one though). In delta one could for instance define a column name that does contain a `.` character under the `columnMapping` feature. In kernel we have some places where we do represent nested coluns as dot separated, but we do already know that we eventually need an array representation.

One reason one might want to specify a nested field, is if we cannot compute stats for some child fields. There we could also just be lenient and simply omit these (actually not sure what we do now :D). Is that your case as well, or do you have others?

Reply:
btw - the fact that `.fields()` does not any parsing / traversing is by design. i.e. in this case we would need to split here...

Reply:
So turns out nested fields are perfectly valid, but they do require some more parsing, as it is also legal to escape field names that contain special characters.

https://github.com/delta-io/delta/blob/4b102d34a2ce881b2a851b4c6cfbf2ab3ab5534f/spark/src/main/scala/org/apache/spark/sql/delta/DeltaConfig.scala#L549-L561

Reply:
i opened #2572 to track this.

Reply:
in our case if nested fields our omitted thats fine we only care about the stats on the root fields for query performance

we only included the nested fields cause they are not nullable fields of nullable parent fields and if we didn't collect them chckpoint parsing would break

also at the time, delta-rs did not respect the stats configuration so we just used the parquet writer properties to only collect stats on what we needed